lib/AppleNotesFolder.rb

```diff
@@ -131,17 +135,35 @@ def sorted_notes
     @notes.sort_by { |note| [note.is_pinned ? 1 : 0, note.modify_time] }.reverse
   end
 
-  def generate_folder_hierarchy_html
+  def to_path
+    path = Pathname.new(@name.tr('/:', '_'))
+    if @parent
+      return @parent.to_path.join(path)
+    end
+    return path
+  end
+
+  def unique_id
```

Comment:
Both AppleNote and AppleNotesFolder inherit from AppleCloudKitRecord. This `unique_id` function could be defined there to only need one copy.

Reply:
Hmm, the implementations are slightly different though; the `AppleNote` one uses `note_id` whereas the `AppleNotesFolder` implementation uses `primary_key`. I see that `AppleCloudKitRecord` doesn't define `uuid` or `primary_key` properties so it might feel a little weird to have a method referencing those properties in the class. What do you think?

Reply:
Fair play, I glossed over that as I was reviewing. What about this:

- Define an `@unique_id` variable on the `AppleCloudKitRecord`, now the subordinate classes have it
- In the initializer of `AppleNotesFolder` and `AppleNote`, set `@unique_id` to be the corresponding value (`note_id` or `primary_key`). 
- Where the `AppleNotesFolder` and `AppleNote` are created (generally in `AppleNoteStore`) check to see if the UUID switch is turned on. If it is, call some other function like `use_uuid` that will then set `@unique_id` to the uuid.

This way you know it always defaults to some value and can easily account for where it needs to be set otherwise. I'm not 100% sold that's the best way, just a spitball as I consider the point you raised.

Reply:
I ended up taking a slightly different approach in the commit I just pushed - let me know if that works for you. I figured since the option has to be plumbed through a few places either way, might as well keep it to the output-related methods since that's what it affects.

Reply:
Yeah, this is fine. I might tweak it in the future if I end up with strong feelings as I think about it, but I see no reason to hold off on merging it the way you've implemented it.